crates/bdk/src/wallet/mod.rs

```diff
@@ -489,17 +475,14 @@ impl fmt::Display for ApplyBlockError {
 #[cfg(feature = "std")]
 impl std::error::Error for ApplyBlockError {}
 
-impl<D> Wallet<D> {
+impl Wallet {
     /// Initialize an empty [`Wallet`].
     pub fn new<E: IntoWalletDescriptor>(
         descriptor: E,
         change_descriptor: Option<E>,
-        db: D,
+        db: Box<dyn PersistBackend<ChangeSet>>,
```

Comment:
Should we hide the `Box` from the caller and call `Box::new` internally? 
```suggestion
        db: impl PersistBackend<ChangeSet> + 'static,
```